### PR TITLE
Upp hafsv1

### LINKED ISSUE #742: merger hafsv1 upp updates to UPP develop branch
Add POT variables, accumulated APCP, ACPCP, NCPCP and updated the CLMR in hafs control files and add a hafs control file which combines hafs_nosat and hafs_sat control files.

### PR DESCRIPTION
**Pull request by LinZhu-NOAA**
Merge hafsv1 changes to UPP develop branch. The changes are adding POT variables, accumulated APCP, ACPCP, NCPCP and updated the CLMR in hafs control files and adding a hafs control file which combines hafs_nosat and hafs_sat control files.

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@LinZhu-NOAA Can you sync your branch up with the UPP develop?

**Comment by LinZhu-NOAA** [comment]
@WenMeng-NOAA I tried the "Sync fork" in my branch, it showed "This branch is not behind the upstream NOAA-EMC:develop". Is there other ways to rsync? Thanks!

**Comment by WenMeng-NOAA** [comment]
@LinZhu-NOAA You might follow branch syncing instructions at https://github.com/NOAA-EMC/UPP/wiki/UPP-Code-Development#update-your-branch-at-your-fork-with-branch-develop-at-authoritative-repository

BTW it appears to me your last syncing was successful.

**Comment by WenMeng-NOAA** [comment]
@LinZhu-NOAA I wonder if you would like to modify the UPP user control text file generation script UPP/parm/makefile with new hafs xml file postcntrl_hafs.xml?

**Comment by LinZhu-NOAA** [comment]
@WenMeng-NOAA Thanks for the instruction. It is great that the syncing works. For the makefile, sure,  I will add the new hafs xml in the makefile and let your know once it is done. Thanks!

**Comment by LinZhu-NOAA** [comment]
@WenMeng-NOAA The parm/makefile is updated. Please check. Thanks!

File: parm/postcntrl_hafs_nosat.xml [hunk 1]
```diff
@@ -92,7 +99,7 @@
        </param>
 
        <param>
-       <shortname>CLWMR_ON_ISOBARIC_SFC</shortname>
+       <shortname>CLMR_ON_ISOBARIC_SFC</shortname>
        <pname>CLWMR</pname>
```
Comment (WenMeng-NOAA) [inline parm/postcntrl_hafs_nosat.xml]: @LinZhu-NOAA The shortname should be CLMR.

File: parm/postcntrl_hafs.xml [hunk 2]
```diff
@@ -0,0 +1,645 @@
...
+       <shortname>ABSV_ON_ISOBARIC_SFC</shortname>
+       <pname>ABSV</pname>
+       <level>200. 500. 700. 1000. 2000. 3000. 5000. 7000. 10000. 12500. 15000. 17500. 20000. 22500. 25000. 27500. 30000. 32500. 35000. 37500. 40000. 42500. 45000.
+47500. 50000. 52500. 55000. 57500. 60000. 62500. 65000. 67500. 70000. 72500. 75000. 77500. 80000. 82500. 85000. 87500. 90000. 92500. 95000. 97500. 100000.</level>
+       <scale>4.0</scale>
+       </param>
+
+       <param>
+       <shortname>CLMR_ON_ISOBARIC_SFC</shortname>
+       <pname>CLWMR</pname>
```
Comment (WenMeng-NOAA) [inline parm/postcntrl_hafs.xml]: @LinZhu-NOAA The shortname should be CLMR.

**Comment by LinZhu-NOAA** [comment]
@WenMeng-NOAA The shortname is changed from "CLWMR" to "CLMR", is the pname should change to "CLMR"? Thanks!

**Comment by WenMeng-NOAA** [comment]
> @WenMeng-NOAA The shortname is changed from "CLWMR" to "CLMR", is the pname should change to "CLMR"? Thanks!

@LinZhu-NOAA Yes, sorry for typo.

**Comment by LinZhu-NOAA** [comment]
@WenMeng-NOAA The pname of CLMR in hafs_nosat.xml and hafs.xml is updated. Please check. Thanks!

**Comment by WenMeng-NOAA** [comment]
@LinZhu-NOAA I am not sure how you generated postxconfig-NT-hafs.txt and postxconfig-NT-hafs_nosat.txt. Can you recreate them via makefile script as:
```
cd UPP/parm
rm postxconfig-NT-hafs.txt, postxconfig-NT-hafs_nosat.txt
make
```
Commit these two files again.

**Comment by LinZhu-NOAA** [comment]
>

**Comment by LinZhu-NOAA** [comment]
> @LinZhu-NOAA I am not sure how you generated postxconfig-NT-hafs.txt and postxconfig-NT-hafs_nosat.txt. Can you recreate them via makefile script as:
> 
> ```
> cd UPP/parm
> rm postxconfig-NT-hafs.txt, postxconfig-NT-hafs_nosat.txt
> make
> ```
> 
> Commit these two files again.

@WenMeng-NOAA Actually, I just modified the old hafs_nosat.txt for adding new variables :) The hafs.txt and hafs_nosat.txt are regenerated by "make".

**Comment by WenMeng-NOAA** [comment]
> > @LinZhu-NOAA I am not sure how you generated postxconfig-NT-hafs.txt and postxconfig-NT-hafs_nosat.txt. Can you recreate them via makefile script as:
> > ```
> > cd UPP/parm
> > rm postxconfig-NT-hafs.txt, postxconfig-NT-hafs_nosat.txt
> > make
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Commit these two files again.
> 
> @WenMeng-NOAA Actually, I just modified the old hafs_nosat.txt for adding new variables :) The hafs.txt and hafs_nosat.txt are regenerated by "make".

@LinZhu-NOAA Right now they look good to me. Thanks!

**Comment by WenMeng-NOAA** [comment]
@LinZhu-NOAA Please see my testing with your new postxconfig-NT-hafs_nosat.txt at /u/wen.meng/ptmp/fv3hafs_2022092800 on dogwood and let me know if you see unexpected changes from your testing.

**Comment by LinZhu-NOAA** [comment]
> @LinZhu-NOAA Please see my testing with your new postxconfig-NT-hafs_nosat.txt at /u/wen.meng/ptmp/fv3hafs_2022092800 on dogwood and let me know if you see unexpected changes from your testing.

@WenMeng-NOAA The result looks fine to me. Thanks!

**Comment by WenMeng-NOAA** [comment]
The UPP RT tests were completed on WCOSS2. A new baseline of hafs is needed with this PR since 54 new fields are added in hafs control files.

**Comment by FernandoAndrade-NOAA** [comment]
UPP RTs have completed on Hera with changes to hafs results HURPRS09.tm00.
@WenMeng-NOAA I did not see any changes in the gfs results we discussed earlier, I'm still discussing with Natalie, but the crtm seems fine on both orion and hercules.

**Comment by WenMeng-NOAA** [comment]
> UPP RTs have completed on Hera with changes to hafs results HURPRS09.tm00. @WenMeng-NOAA I did not see any changes in the gfs results we discussed earlier, I'm still discussing with Natalie, but the crtm seems fine on both orion and hercules.

@FernandoAndrade-NOAA It's great your testing on Hera has no issue from simulated satellite data generation.

**Review by FernandoAndrade-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
The UPP RT tests were completed on Orion.

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.

**Pull request merged**